### PR TITLE
Ensure we perform updates to Document sequentially

### PR DESCRIPTION
**Pull request by DonJayamanne**
@davidanthoff 
Not sure I need an issue when submitting changes to the feature branch.
Same with updates to the change log.

Let me know if these are required.

### CODE REVIEW

File: src/notebook/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Depending on the sequence its possible for some of the updates to get lost.
+ *
+ * Excellent example:
+ * Assume we perform the following updates without awaiting on the promise.
+ * Without awaiting, its very easy to replicate issues where the output is never displayed.
+ * - We update execution count
+ * - We update output
+ * - We update status after completion
+ */
+const pendingCellUpdates = new WeakMap<NotebookDocument, Promise<unknown>>()
```
Comment (DonJayamanne) [inline src/notebook/helpers.ts:17]: Hopefully by next week all of these work arounds will be a thing of the past, VSCode is updating their API (changes to how execution is handled and output is updated).

Comment (DonJayamanne) [reply src/notebook/helpers.ts:17]: This is one of the reasons why outputs might not appear at times,

Comment (DonJayamanne) [reply src/notebook/helpers.ts:17]: julia

**Comment by davidanthoff** [comment]
> Not sure I need an issue when submitting changes to the feature branch.

No need for issues, we can just move quickly on all of this without much process as all, especially given that we do all work on a branch.

> Same with updates to the change log.

Also: no need for that.

**Pull request merged**